open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+        return validationResponse;
+    }
+
+    /**
+     * Validates the Currency in periodic limits in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateCurrencyPeriodicLimit(JSONObject controlParameters) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+        JSONObject currency = validateAmountCurrencyPeriodicLimits((JSONArray) periodicLimits,
+                ConsentExtensionConstants.CURRENCY);
+        if (!(Boolean.parseBoolean(currency.
+                getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return currency;
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+
+    /**
+     * Validates the Amount in periodic limits in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateAmountPeriodicLimit(JSONObject controlParameters) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+        JSONObject amount = validateAmountCurrencyPeriodicLimits((JSONArray) periodicLimits,
+                ConsentExtensionConstants.AMOUNT);
+        if (!(Boolean.parseBoolean(amount.
+                getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return amount;
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+
+    /**
+     * Validates the date-time parameters in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response. If the date-time parameters are valid,
+     * it sets the "IS_VALID" field to true; otherwise, it contains an error response.
+     */
+    public static JSONObject validateParameterDateTime(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        // Check if ValidToDateTime key is present
+        if (controlParameters.containsKey(ConsentExtensionConstants.VALID_TO_DATE_TIME)) {
+            // Get and validate ValidToDateTime
+            Object validateToDateTime = controlParameters.get(ConsentExtensionConstants.VALID_TO_DATE_TIME);
+            JSONObject toDateTimeValidation = isValidDateTimeObject(validateToDateTime);
```

Comment:
Here also toDateTimeValidation  means the validationResponse for `validToDateTime` right? , shall we use naming to indicate that here as well? Ex :  `validToDateTimeValidationResponse` WDUT? , Check other places too

Reply:
Fixed in [876c68f]